imap-proto/src/types.rs

```diff
@@ -88,6 +89,37 @@ impl From<u32> for UidSetMember {
     }
 }
 
+impl<'a> ResponseCode<'a> {
+    pub fn into_owned(self) -> ResponseCode<'static> {
+        match self {
+            ResponseCode::Alert => ResponseCode::Alert,
+            ResponseCode::BadCharset(v) => ResponseCode::BadCharset(v.map(|vs| {
+                vs.into_iter()
+                    .map(Cow::into_owned)
+                    .map(Cow::Owned)
```

Comment:
Surely we we can do better than `map().map()`?

Reply:
I actually found the former a little nicer to read since it doesn't require the closure, but will change :+1:

Reply:
Added a helper function — much nicer now!